cli/src/main.rs

```diff
@@ -1,35 +1,91 @@
+use std::env;
 use std::{fs, path::PathBuf};
 
 use argh::FromArgs;
-use wasmstation::backend::WasmerBackend;
-use wasmstation::console::Console;
+use wasmstation::{backend::WasmerBackend, console::Console};
 
 #[derive(FromArgs)]
 #[argh(description = "Run wasm4 compatible games.")]
 struct Args {
-    #[argh(positional)]
-    path: PathBuf,
-    #[argh(
-        option,
-        short = 's',
-        default = "3",
-        description = "scale factor for the window"
-    )]
-    display_scale: u32,
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+#[derive(FromArgs)]
+#[argh(subcommand)]
+enum Subcommand {
+    Run(Run),
+    Create(Create),
 }
 
 fn main() {
     let args: Args = argh::from_env();
 
     pretty_env_logger::init();
 
+    match args.subcommand {
+        Subcommand::Run(args) => run(args),
+        Subcommand::Create(args) => create(args),
+    }
+}
+
+/// Run a WASM-4 game.
+#[derive(FromArgs)]
+#[argh(subcommand, name = "run")]
+struct Run {
+    #[argh(positional)]
+    path: PathBuf,
+    /// default scale factor for the window
+    #[argh(option, short = 's', default = "3")]
+    display_scale: u32,
+}
+
+fn run(args: Run) {
     let wasm_bytes = fs::read(&args.path).expect("failed to read game");
-    let console = Console::new();
 
     wasmstation::launch(
-        WasmerBackend::new(&wasm_bytes, &console).unwrap(),
+        WasmerBackend::new(&wasm_bytes, &Console::new()).unwrap(),
         &args.path,
         args.display_scale,
     )
     .unwrap();
 }
+
+/// Create an executable from a WASM-4 game.
```

Comment:
I think this should explain a bit more what it actually does. To me the help command printed this:

```
Usage: wasmstation create <path> [-s <display-scale>] [-t <name>]

Create an executable from a WASM-4 game.

Positional Arguments:
  path

Options:
  -s, --display-scale
                    scale factor of the window
  -t, --name        name of the game
  --help            display usage information
```

My open questions assuming I'm a new user when reading the above:
* What does it do exactly - does it create a native executable? (no it doesn't, it creates a project)
* Where does it create the project? (in the current directory, but I only notice that after I ran it once and maybe polluted my cwd)
* what does `path` point to? (knowing it creates a project, my first assumption was it's the target directory, but it's the path to the cart.

I suggest to change the usage information so that it looks like this:

```
Usage: wasmstation create <path> [-s <display-scale>] [-t <name>]

Create a Rust project for wrapping a WASM-4 game into a native executable in the current directory.

Positional Arguments:
  path-to-cart
                    path leading to the game's cartridge file (e.g. /path/to/watris.wasm)

[..]
```

Reply:
Thank you, this is another good point. I believe my vague subcommand description was left over from my old implementation which hid the project from the user and only provided the resulting executable.

In my most recent commit the description of the subcommand looks like this:

```
Usage: wasmstation create <cart> [-s <display-scale>]

Create a Rust project for wrapping a WASM-4 game into a native executable in the current directory.

Positional Arguments:
  cart              path leading to the game's cartridge file (e.g.
                    /path/to/cart.wasm)

Options:
  -s, --display-scale
                    default scale factor of the window
  --help            display usage information
  ```